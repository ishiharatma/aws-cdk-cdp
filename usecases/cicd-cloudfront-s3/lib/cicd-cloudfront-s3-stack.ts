import * as cdk from 'aws-cdk-lib';
import { StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { WebPipelineConstruct } from '../../common/lib/construct-pipeline-s3web'

import * as interfaceDef from '../../common/interface/index';

interface CicdCloudFrontS3StackProps extends StackProps {
  input: interfaceDef.WebPipelineConstructProps;
}

export class CicdCloudfrontS3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CicdCloudFrontS3StackProps) {
    super(scope, id, props);

    new WebPipelineConstruct(this, 'Default', props.input);

  }
}
