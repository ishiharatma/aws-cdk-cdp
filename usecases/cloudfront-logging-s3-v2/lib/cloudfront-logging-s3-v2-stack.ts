import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  aws_logs as logs,
} from 'aws-cdk-lib';
interface CloudFrontLoggingS3V2StackProps extends cdk.StackProps {
  readonly pjName: string;
  readonly envName: string;
  /*
    @param distributionId: string;
    @required
  */
    readonly distributionId: string;
    /*
      @param logPrefix?: string;
      @default /{DistributionId}/{yyyy}/{MM}/{dd}/{HH}
    */
    readonly logPrefix?: string;
    /*
      @param HiveCompatiblePath?: boolean;
      @default false
    */
    readonly HiveCompatiblePath?: boolean;
    /*
      @param logsDeliverySourceFormat?:string;
      see: https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutDeliveryDestination.html#API_PutDeliveryDestination_RequestSyntax
      @default json
    */
    readonly logsDeliveryDestinationFormat?:string;
    /*
      @param loggingBucketArn: string;
      @required
    */
    readonly loggingBucketArn: string;
}

export class CloudFrontLoggingS3V2Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CloudFrontLoggingS3V2StackProps) {
    super(scope, id, props);
    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;

    // logging v2
    // see: https://docs.aws.amazon.com/ja_jp/ja_jp/AmazonCloudFront/latest/DeveloperGuide/standard-logging.html
    // Logs: Delivery Source
    const logsDeliverySource = new logs.CfnDeliverySource(this, 'LogsDeliverySource', {
      name: `${props.pjName}-${props.envName}-cloudfront-delivery-source`,
      resourceArn: `arn:aws:cloudfront::${accountId}:distribution/${props.distributionId}`,
      logType: 'ACCESS_LOGS'
    });
    // Logs: Delivery Destination
    const logsDeliveryDestination = new logs.CfnDeliveryDestination(this, 'LogsDeliveryDestination', {
      name: `${props.pjName}-${props.envName}-cloudfront-delivery-destination`,
      destinationResourceArn: `${props.loggingBucketArn}`,///AWSLogs/${accountId}/CloudFront`,
      outputFormat: props.logsDeliveryDestinationFormat ?? 'json'
    });
    // Logs: Delivery
    const logsDelivery = new logs.CfnDelivery(this, 'LogsDelivery', {
      deliverySourceName: `${props.pjName}-${props.envName}-cloudfront-delivery-source`,
      deliveryDestinationArn: logsDeliveryDestination.attrArn,
      recordFields: [
        'timestamp', 'DistributionId', 'date', 'time', 'x-edge-location', 
        'sc-bytes', 'c-ip', 'cs-method', 'cs(Host)', 'cs-uri-stem', 
        'sc-status', 'cs(Referer)', 'cs(User-Agent)', 'cs-uri-query', 
        'cs(Cookie)', 'x-edge-result-type', 'x-edge-request-id', 
        'x-host-header', 'cs-protocol', 'cs-bytes', 'time-taken', 
        'x-forwarded-for', 'ssl-protocol', 'ssl-cipher', 
        'x-edge-response-result-type', 'cs-protocol-version', 'fle-status', 
        'fle-encrypted-fields', 'c-port', 'time-to-first-byte', 
        'x-edge-detailed-result-type', 'sc-content-type', 'sc-content-len', 
        'sc-range-start', 'sc-range-end', 'timestamp(ms)', 'origin-fbl', 
        'origin-lbl', 'asn'
      ],
      s3SuffixPath: props.logPrefix ?? '/v2/{DistributionId}/{yyyy}/{MM}/{dd}/{HH}',
      s3EnableHiveCompatiblePath: props.HiveCompatiblePath ?? false
    });
    // 依存関係の設定
    logsDelivery.addDependency(logsDeliverySource);
    logsDelivery.addDependency(logsDeliveryDestination);

  }
}
