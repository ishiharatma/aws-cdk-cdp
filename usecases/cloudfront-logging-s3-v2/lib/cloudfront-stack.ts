import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  aws_s3 as s3,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins as origins, 
  aws_s3_deployment as s3deploy,
  aws_logs as logs,
} from 'aws-cdk-lib';
import * as path from 'path';
interface CloudFrontStackProps extends cdk.StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly isAutoDeleteObject?: boolean;
}

export class CloudFrontStack extends cdk.Stack {
  public readonly distribution:cloudfront.IDistribution;
  public readonly loggingBucket:s3.IBucket;
  constructor(scope: Construct, id: string, props: CloudFrontStackProps) {
    super(scope, id, props);
    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;
    const contentsPath:string = path.join(__dirname, '../src/static-site');

    // create logging bucket for cloudfront
    this.loggingBucket = new s3.Bucket(this, 'LoggingBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    }); 
    // create static website bucket
    const staticWebSiteBucket = new s3.Bucket(this, 'StaticWebSiteBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    });

    const responseHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
      this,
      'ResponseHeadersPolicy',
      {
        securityHeadersBehavior: {
          contentTypeOptions: { override: true },
          frameOptions: {
            frameOption: cloudfront.HeadersFrameOption.DENY,
            override: true,
          },
          referrerPolicy: {
            referrerPolicy: cloudfront.HeadersReferrerPolicy.SAME_ORIGIN,
            override: true,
          },
          strictTransportSecurity: {
            accessControlMaxAge: cdk.Duration.seconds(63072000),
            includeSubdomains: true,
            preload: true,
            override: true,
          },
          xssProtection: {
            protection: true,
            modeBlock: true,
            override: true,
          },
        },
        customHeadersBehavior: {
          customHeaders: [
            {
              header: 'Cache-Control',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'pragma',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'server',
              value: '',
              override: true,
            },
          ],
        },
      }
    );
    // create cloudfront distribution
    this.distribution = new cloudfront.Distribution(this, 'MyDistribution', {
      comment: 'My distribution',
      enabled: true,
      httpVersion: cloudfront.HttpVersion.HTTP2_AND_3,
      // see: https://docs.aws.amazon.com/ja_jp/AmazonCloudFront/latest/DeveloperGuide/PriceClass.html
      priceClass: cloudfront.PriceClass.PRICE_CLASS_200, //or PRICE_CLASS_ALL
      minimumProtocolVersion: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
      enableIpv6: false,
      defaultRootObject: "index.html",

      defaultBehavior: {
        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD,
        cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        compress: true,
        origin: origins.S3BucketOrigin.withOriginAccessControl(staticWebSiteBucket),
        originRequestPolicy: cloudfront.OriginRequestPolicy.CORS_S3_ORIGIN,
        responseHeadersPolicy
      },
    });

    new s3deploy.BucketDeployment(this, 'IndexHtmlDeploy', {
      sources: [
        s3deploy.Source.asset(contentsPath),
      ],
      destinationBucket: staticWebSiteBucket,
      distribution: this.distribution,
      distributionPaths: ["/*"],
    });

  }
}
