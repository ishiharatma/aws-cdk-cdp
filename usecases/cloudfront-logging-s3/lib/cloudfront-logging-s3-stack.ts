import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  aws_s3 as s3,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins as origins, 
  aws_s3_deployment as s3deploy,
} from 'aws-cdk-lib';
import * as path from 'path';
interface CloudfrontLoggingS3StackProps extends cdk.StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly isAutoDeleteObject?: boolean;
}

export class CloudfrontLoggingS3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CloudfrontLoggingS3StackProps) {
    super(scope, id, props);

    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;
    const contentsPath:string = path.join(__dirname, '../src/static-site');

    // create logging bucket for cloudfront
    const loggingBucket = new s3.Bucket(this, 'LoggingBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    }); 
    // create static website bucket
    const staticWebSiteBucket = new s3.Bucket(this, 'StaticWebSiteBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const responseHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
      this,
      'ResponseHeadersPolicy',
      {
        securityHeadersBehavior: {
          contentTypeOptions: { override: true },
          frameOptions: {
            frameOption: cloudfront.HeadersFrameOption.DENY,
            override: true,
          },
          referrerPolicy: {
            referrerPolicy: cloudfront.HeadersReferrerPolicy.SAME_ORIGIN,
            override: true,
          },
          strictTransportSecurity: {
            accessControlMaxAge: cdk.Duration.seconds(63072000),
            includeSubdomains: true,
            preload: true,
            override: true,
          },
          xssProtection: {
            protection: true,
            modeBlock: true,
            override: true,
          },
        },
        customHeadersBehavior: {
          customHeaders: [
            {
              header: 'Cache-Control',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'pragma',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'server',
              value: '',
              override: true,
            },
          ],
        },
      }
    );
    // create cloudfront distribution
    const distribution = new cloudfront.Distribution(this, 'MyDistribution', {
      comment: 'My distribution',
      enabled: true,
      httpVersion: cloudfront.HttpVersion.HTTP2_AND_3,
      // see: https://docs.aws.amazon.com/ja_jp/AmazonCloudFront/latest/DeveloperGuide/PriceClass.html
      priceClass: cloudfront.PriceClass.PRICE_CLASS_200, //or PRICE_CLASS_ALL
      minimumProtocolVersion: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
      enableIpv6: false,
      defaultRootObject: "index.html",
      // logging (legacy)
      // see: https://docs.aws.amazon.com/ja_jp/ja_jp/AmazonCloudFront/latest/DeveloperGuide/standard-logging-legacy-s3.html
      enableLogging: true,
      logBucket: loggingBucket,
      logFilePrefix: undefined,

      defaultBehavior: {
        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD,
        cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        compress: true,
        origin: origins.S3BucketOrigin.withOriginAccessControl(staticWebSiteBucket),
        originRequestPolicy: cloudfront.OriginRequestPolicy.CORS_S3_ORIGIN,
        responseHeadersPolicy
      },
    });
    new s3deploy.BucketDeployment(this, 'IndexHtmlDeploy', {
      sources: [
        s3deploy.Source.asset(contentsPath),
      ],
      destinationBucket: staticWebSiteBucket,
      distribution: distribution,
      distributionPaths: ["/*"],
    });
  }
}
