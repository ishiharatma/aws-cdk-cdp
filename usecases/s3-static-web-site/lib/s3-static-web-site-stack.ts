import * as cdk from 'aws-cdk-lib';
import {
  StackProps,
  aws_s3 as s3,
  aws_iam as iam,
} from 'aws-cdk-lib';
import * as path from 'path';
import { Construct } from 'constructs';
import { BucketConstruct } from '../../common/lib/construct-bucket';
import { BucketAddPolicyConstruct } from '../../common/lib/construct-bucket-add-bucketpolicy';

interface S3StaticWebSiteStackProps extends StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly isAutoDeleteObject?: boolean;
  readonly allowedIpV4AddressRanges: string[];
}

export class S3StaticWebSiteStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: S3StaticWebSiteStackProps) {
    super(scope, id, props);
    // Bucket for static web site Access Log
    const websiteAccessLogsBucket = new BucketConstruct(this,'WebsiteLogsBucket',{
      pjName: props.pjName,
      envName: props.envName,
      bucketSuffix: 'static-website-logs',
      accessControl: s3.BucketAccessControl.LOG_DELIVERY_WRITE,
      isAutoDeleteObject: props.isAutoDeleteObject,
      lifecycleRules: [
        {
          expirationDays: 90,
          abortIncompleteMultipartUploadAfterDays: 7,
        }
      ]
    });
    // Bucket for static web site
    const websiteBucket = new BucketConstruct(this,'WebsiteBucket',{
      pjName: props.pjName,
      envName: props.envName,
      bucketSuffix: 'static-website',
      isAutoDeleteObject: props.isAutoDeleteObject,
      s3ServerAccessLogBucketConstruct: websiteAccessLogsBucket,
      logFilePrefix: '',
      isPublicReadAccess: props.allowedIpV4AddressRanges.length > 0 ? false: true,
      isStaticWebSite: true,
      contentsPath: path.join(__dirname, '../../common/src/static-site/web'),
    });

    if (props.allowedIpV4AddressRanges.length > 0) {
      // Bucket Policy
      const bucketPolicy = new iam.PolicyStatement({
        effect: iam.Effect.ALLOW,
        actions: ['s3:GetObject'],
        principals: [new iam.ArnPrincipal('*')],
        resources: [
          websiteBucket.bucket.arnForObjects('*'),
        ],
        conditions: {
          'IpAddress': {
            'aws:SourceIp': props.allowedIpV4AddressRanges
          }
        }
      });
  
      new BucketAddPolicyConstruct(this, 'AddBucketPolicy', {
        pjName: props.pjName,
        envName: props.envName,
        targetS3: websiteBucket,
        s3BucketPolicyStatementJson: bucketPolicy.toJSON(),
      });  
    };

    new cdk.CfnOutput(this, 'BucketWebsiteUrl', {
      value: websiteBucket.bucket.bucketWebsiteUrl,
      description: 'URL for website hosted on S3'
    });

  }
}
