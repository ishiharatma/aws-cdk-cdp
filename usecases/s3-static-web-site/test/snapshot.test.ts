import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { App } from 'aws-cdk-lib';

import { S3StaticWebSiteStack } from '../lib/s3-static-web-site-stack';

const defaultEnv = {
    account: '123456789012',
    region: 'ap-northeast-1',
};
const pjName: string = 'snapshot';
const envName: string = 'test';
const app = new App();

test('snapshot validation test',() =>{
    const stack = new S3StaticWebSiteStack(app, 'VPCStack', {
        pjName: pjName,
        envName: envName,
        description: '',
        isAutoDeleteObject: true,
        allowedIpV4AddressRanges: ['0.0.0.0/0'],
        env: defaultEnv,
        terminationProtection: true, // Enabling deletion protection
      });
    // add tag
    cdk.Tags.of(app).add('Project', pjName);
    cdk.Tags.of(app).add('Environment', envName);
    // test with snapshot
    expect(Template.fromStack(stack)).toMatchSnapshot();

})