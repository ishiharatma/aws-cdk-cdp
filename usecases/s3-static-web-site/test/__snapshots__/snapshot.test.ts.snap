// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot validation test 1`] = `
{
  "Description": "",
  "Outputs": {
    "BucketWebsiteUrl": {
      "Description": "URL for website hosted on S3",
      "Value": {
        "Fn::GetAtt": [
          "WebsiteBucketS3BucketFE46300C",
          "WebsiteURL",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "c6358465bf49dfae556bb430bf9c81fa578c221b82c308e3707901b1dd654762.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_CA_BUNDLE": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
          },
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "WebsiteBucketS3FileUploadAwsCliLayer0772B767",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.11",
        "Tags": [
          {
            "Key": "Environment",
            "Value": "test",
          },
          {
            "Key": "Project",
            "Value": "snapshot",
          },
        ],
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Environment",
            "Value": "test",
          },
          {
            "Key": "Project",
            "Value": "snapshot",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-123456789012-ap-northeast-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-123456789012-ap-northeast-1/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteBucketS3BucketFE46300C",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteBucketS3BucketFE46300C",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "faa95a81ae7d7373f3e1f242268f904eb748d8d0fdd306e8a6fe515a1905a7d6.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "WebsiteLogsBucketS3BucketA5B66D15",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "WebsiteBucketS3BucketAutoDeleteObjectsCustomResourceA0F5CD3D": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "WebsiteBucketS3BucketPolicyF2157EA2",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "WebsiteBucketS3BucketFE46300C",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketS3BucketFE46300C": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "snapshot-test-static-website-123456789012",
        "LoggingConfiguration": {
          "DestinationBucketName": {
            "Ref": "WebsiteLogsBucketS3BucketA5B66D15",
          },
          "LogFilePrefix": "",
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "aws-cdk:cr-owned:723b95f1",
            "Value": "true",
          },
          {
            "Key": "Environment",
            "Value": "test",
          },
          {
            "Key": "Project",
            "Value": "snapshot",
          },
        ],
        "WebsiteConfiguration": {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketS3BucketPolicyF2157EA2": {
      "Properties": {
        "Bucket": {
          "Ref": "WebsiteBucketS3BucketFE46300C",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteBucketS3BucketFE46300C",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteBucketS3BucketFE46300C",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Condition": {
                "IpAddress": {
                  "aws:SourceIp": [
                    "0.0.0.0/0",
                  ],
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "WebsiteBucketS3BucketFE46300C",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "WebsiteBucketS3FileUploadAwsCliLayer0772B767": {
      "Properties": {
        "Content": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "1c5a2ab1d1e53b0672a644454aab3dbb258ccd0079c92ad0e23b95b2c2079f70.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "WebsiteBucketS3FileUploadCustomResource64803444": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "WebsiteBucketS3BucketFE46300C",
        },
        "OutputObjectKeys": true,
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
        ],
        "SourceObjectKeys": [
          "133b62e6550924b60dea459dfcdde5c077c0d2e8821f00733b577aa2f05b4fe5.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteLogsBucketS3BucketA5B66D15": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "snapshot-test-static-website-logs-123456789012",
        "LifecycleConfiguration": {
          "Rules": [
            {
              "AbortIncompleteMultipartUpload": {
                "DaysAfterInitiation": 7,
              },
              "ExpirationInDays": 90,
              "ExpiredObjectDeleteMarker": false,
              "Id": "Delete-After-90Days",
              "Status": "Enabled",
            },
          ],
        },
        "OwnershipControls": {
          "Rules": [
            {
              "ObjectOwnership": "ObjectWriter",
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "Environment",
            "Value": "test",
          },
          {
            "Key": "Project",
            "Value": "snapshot",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteLogsBucketS3BucketAutoDeleteObjectsCustomResource8C7D16B9": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "WebsiteLogsBucketS3BucketPolicy1C0BD55C",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "WebsiteLogsBucketS3BucketA5B66D15",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteLogsBucketS3BucketPolicy1C0BD55C": {
      "Properties": {
        "Bucket": {
          "Ref": "WebsiteLogsBucketS3BucketA5B66D15",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteLogsBucketS3BucketA5B66D15",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteLogsBucketS3BucketA5B66D15",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
